libs/agno/agno/session/agent.py

```diff
@@ -151,12 +151,21 @@ def get_messages_from_last_n_runs(
                 continue
 
             for message in run_response.messages or []:
-                # Skip messages with specified role
-                if skip_role and message.role == skip_role:
-                    continue
                 # Skip messages that were tagged as history in previous runs
                 if hasattr(message, "from_history") and message.from_history and skip_history_messages:
                     continue
+
+                # Skip messages with specified role, but ONLY for role="system"
+                # The original intent of skip_role was to avoid duplicate system messages.
+                # However, when system_message_role is set to roles like "user", "assistant",
+                # or custom roles, we should NOT filter those messages from history because:
+                # 1. The system message is regenerated fresh for each run anyway
+                # 2. Messages with those roles in history are legitimate conversation messages
+                # 3. Filtering them would break conversation continuity
+                # Therefore, we only apply skip_role filtering for the traditional "system" role.
+                if skip_role and message.role == skip_role and skip_role == "system":
```

Comment:
I don't think we should do it like this, because it breaks the interface of the function. 
Lets rather go to where this function is called and set `skip_role` to `system_message_role`?